### DISCUSSION TITLE
Vite 4 + xState - freeze browser on production

### DISCUSSION THREAD

**Discussion by PiotrGrobelak** [statelyai/xstate#3751] [Q&A] [answered]
In my work, I try to migrate to Vite 4 but after an upgrade and building the application with production mode every process where I use xState stopped working and freeze the browser tab. The machine started and initialize the state and for example after the second promise stopped with a frozen browser without errors and some notifications.  I don't know if it is a problem production build with Vite 4 or xState implementation. With Vite 3 I didn't have problems on production. In development mode, everything works fine.

Currently, I can't deliver source code but in the future, I would like to test this problem on the external repository. 

Maybe anyone has a similar problem?

**Comment by davidkpiano** [comment, marked as answer]
That's strange; nothing in XState should cause that to happen.

Is it reproducible with any other XState code?

**Reply by PiotrGrobelak** [reply]
In the next week, I create a clean repo with Vite 4 + Vue 3 + TS + xState and paste some code with logic from my work project to check if also is a problem with the frozen browser or works fine.

**Reply by PiotrGrobelak** [reply]
Hi. I tested a little my own repo with Vite 4 + Vue 3 + xState and everything is working fine. Probably the problem is in my production code. I have some bugs or bad implementation rest of the code. 

My link for tests: https://github.com/PiotrGrobelak/vite-4-xState-test but I think that it is not necessary to check this repo.

Close this discussion.
